### ISSUE TITLE
Error on unbind function in Client class

### ISSUE THREAD

**Issue by CatieTort** [ldapts/ldapts#158]
I'm getting an error on this line of code, 

**Error messages:**
A computed property name in an ambient context must refer to an expression whose type is a literal type or a 'unique symbol' type.

Property 'asyncDispose' does not exist on type 'SymbolConstructor'.

```
unbind(): Promise<void>;
    [Symbol.asyncDispose](): Promise<void>;
```

How should this be updated?

**Comment by ayZagen** [comment]
which typescript version are you using ?

**Comment by ayZagen** [comment]
Can you please add the lines below somewhere in your code before creating ldap client? I believe your typescript version is below 5.2 and you need a polyfill.

```ts
Symbol.asyncDispose ??= Symbol("Symbol.asyncDispose");
```

**Comment by CatieTort** [comment]
Yep that is probably the issue, this is what it says in my package.json 

` "typescript": "^4.5.5"`

**Comment by jgeurts** [comment]
I'll get the fix pushed out shortly

**Comment by CatieTort** [comment]
Still gives the same error a few lines below for me. in the Client.ts file

` public [Symbol.asyncDispose](): Promise<void> {
    return this.unbind();
  }
`

**Comment by ayZagen** [comment]
@CatieTort I tried it with `typescript@4.5.5` and I couldn't reproduce the issue. Honestly, I couldn't reproduce it even without the polyfill. Can you share a minimal reproduction ?

**Comment by CatieTort** [comment]
I'm on node@17.0.17 and my code was using ldapjs but the library was deprecated and I switched over to ldapts in my angular project but now nothing seems to be working even though 90% of the code is the same. 

I'm not getting any errors in my code I just see this in my complier when I run ts.

`node_modules/ldapts/dist/index.d.ts:1023:5 - error TS1165: A computed property name in an ambient context must refer to an expression whose type is a literal type or a 'unique symbol' type.

1023     [Symbol.asyncDispose](): Promise<void>;
         ~~~~~~~~~~~~~~~~~~~~~

node_modules/ldapts/dist/index.d.ts:1023:13 - error TS2339: Property 'asyncDispose' does not exist on type 'SymbolConstructor'.

1023     [Symbol.asyncDispose](): Promise<void>;
                 ~~~~~~~~~~~~`

When I tried a modifyDN function to deactivate a user I get a socket error, not sure if it's connected or something seperate.

` unhandledRejection (see full error below):  Socket error. Message type: ModifyDNRequest (0x6c)
read ETIMEDOUT
Error: Socket error. Message type: ModifyDNRequest (0x6c)
read ETIMEDOUT
    at Socket.socketError (/Users/ctortorella/Documents/LJI/ljiapps-api/node_modules/ldapts/dist/index.cjs:3549:13)
    at Socket.emit (node:events:390:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)`

**Comment by ayZagen** [comment]
Hi @CatieTort, you should set `"skipLibCheck": true` option in your `tsconfig` file as this is not a runtime error but compile time error.

**Comment by ayZagen** [comment]
The error you are receiving in `modifyDn` is related to your ldap connection. try to increase `timeout` and `connectTimeout` options in your Client constructor.

**Comment by ayZagen** [comment]
Closed by #159 and #160
